### PR TITLE
TX_POWER is an error

### PR DESCRIPTION
**Pull request by mawdegroot**
TX_POWER is an error when there is no lower or equal power level available.
TX_POWER is a warning when there is a lower or equal power level available.

### CODE REVIEW

**Comment by ke6jjj** [comment]
For context, this PR fixes a bug in the packet forwarder where it accidentally transmits a packet at a higher power than requested when it ought to have returned an error condition instead. This is a bug in the power LUT table processing and how it handles the condition of falling off of the edge of the table.

**Comment by mawdegroot** [comment]
Fixed a final bug where it also made `TX_POWER` an error when there was a lower value available. The following was obtained using a global_conf with values between 16 and 10 dBm.

```
(miner@127.0.0.1)41> gen_server:call(miner_lora_light, {send, <<"...">>, immediate, fun(_) -> 867.9 end, "SF11BW125", 20, false, undefined}, 11000).
{warning,{tx_power_corrected,16}}
(miner@127.0.0.1)42> gen_server:call(miner_lora_light, {send, <<"...">>, immediate, fun(_) -> 867.9 end, "SF11BW125", 6, false, undefined}, 11000).
{error,bad_tx_power}
(miner@127.0.0.1)43> gen_server:call(miner_lora_light, {send, <<"...">>, immediate, fun(_) -> 867.9 end, "SF11BW125", 16, false, undefined}, 11000).
ok
```

**Comment by mawdegroot** [comment]
Rebased and retested.

File: packet_forwarder/src/lora_pkt_fwd.c [hunk 1]
```diff
@@ -1140,6 +1140,10 @@ static int send_tx_ack(uint8_t token_h, uint8_t token_l, enum jit_error_e error,
             buff_index += 9;
             break;
         case JIT_ERROR_TX_POWER:
+            memcpy((void *)(buff_ack + buff_index), (void *)"\"TX_POWER\"", 10);
```
Comment (JayKickliter) [inline packet_forwarder/src/lora_pkt_fwd.c:1143]: Those `(void *)` aren't needed, but you probably already know that and are just following the original code's conventions.

**Review by JayKickliter — APPROVED** [review]

**Review by JayKickliter — APPROVED** [review]

**Pull request merged**